apps/designer/app/designer/shared/client-settings/settings.ts

```diff
@@ -0,0 +1,51 @@
+import * as values from "./values";
+import { useSubscribe } from "@webstudio-is/sdk";
+
+type Name = keyof typeof values;
+type Value = typeof values[Name]["values"][number];
+type Settings = Partial<Record<Name, Value>>;
+
+const namespace = "__webstudio_user_settings__";
+
+const read = (): Settings => {
+  try {
+    const settings = localStorage.getItem(namespace);
+    if (settings === null) return {};
+    return JSON.parse(settings);
+  } catch (error) {
+    // We don't need to handle this one.
+  }
+  return {};
+};
```

Comment:
A test would be great. Also, just swallowing errors doesn't seem to be a good idea. Should those at least be logged and sent to Sentry?

Reply:
That test would be useless, because localStorage would be mocked or polyfilled and otherwise there is nothing in this function to test.

Reply:
My strategy for tests is when there is complex logic or a bug already happened to as a regression test. We are not writing a library and testing every function is not viable or useful. Once we get to e2e tests, they will be able to cover real user flows.

Reply:
> This will only throw if basic localStorage is not available

It will also throw if JSON.parse won't work.

Reply:
right, but that can only mean the data in settings is corrupt, invalid json, what can we do about it except of ignoring it as if it doesn't exist?

Reply:
I guess reporting it to us so we know about this?

Reply:
done